packages/gitbook/src/middleware.ts

```diff
@@ -386,13 +410,31 @@ async function lookupSpaceInSingleMode(url: URL): Promise<LookupResult> {
     };
 }
 
+/**
+ * GITBOOK_MODE=proxy
+ * When proxying a site on a different base URL.
+ */
+async function lookupSiteInProxy(request: NextRequest, url: URL): Promise<LookupResult> {
+    const rawSiteUrl = request.headers.get('x-gitbook-site-url');
+    if (!rawSiteUrl) {
+        throw new Error(
+            `Missing x-gitbook-site-url header. It should be passed when using GITBOOK_MODE=proxy.`,
+        );
+    }
+
+    const siteUrl = new URL(rawSiteUrl);
+    siteUrl.pathname = joinPath(siteUrl.pathname, url.pathname);
```

Comment:
Like I said above, I think it makes more sense to compute this `siteUrl` in `getInputURL` for `proxy` mode and this function just accepts it. Right now `x-gitbook-site-url` is being utilized in more than 1 place (getInputURL and this) and I don't think that should be the case